src/sundials_matrix_wrapper.cpp

```diff
@@ -1,126 +1,83 @@
 #include <amici/sundials_matrix_wrapper.h>
+#include <amici/exception.h>
 
 #include <utility>
 #include <new> // bad_alloc
 
-#include <sundials/sundials_dense.h> // DenseCopy
-
 namespace amici {
 
-SlsMatWrapper::SlsMatWrapper(int M, int N, int NNZ, int sparsetype)
-    : matrix(SparseNewMat(M, N, NNZ, sparsetype))
+SUNMatrixWrapper::SUNMatrixWrapper(int M, int N, int NNZ, int sparsetype)
+    : matrix(SUNSparseMatrix(M, N, NNZ, sparsetype))
 {
     if(NNZ && !matrix)
         throw std::bad_alloc();
 }
-
-SlsMatWrapper::SlsMatWrapper(SlsMat mat)
-    : matrix(mat)
-{}
-
-SlsMatWrapper::~SlsMatWrapper()
-{
-    if(matrix)
-        SparseDestroyMat(matrix);
-}
-
-SlsMatWrapper::SlsMatWrapper(const SlsMatWrapper &other)
-{
-    if(!other.matrix)
-        return;
-
-    matrix = SparseNewMat(other.matrix->M,
-                          other.matrix->N,
-                          other.matrix->NNZ,
-                          other.matrix->sparsetype);
-    if(!matrix)
-        throw std::bad_alloc();
-
-    SparseCopyMat(other.matrix, matrix);
-}
-
-SlsMatWrapper::SlsMatWrapper(SlsMatWrapper &&other) noexcept
-{
-    std::swap(matrix, other.matrix);
-
-}
-
-SlsMatWrapper &SlsMatWrapper::operator=(const SlsMatWrapper &other)
-{
-    return *this = SlsMatWrapper(other);
-}
-
-SlsMatWrapper &SlsMatWrapper::operator=(SlsMatWrapper &&other) noexcept
-{
-    std::swap(matrix, other.matrix);
-    return *this;
-}
-
-realtype *SlsMatWrapper::data() {
-    if(matrix)
-        return matrix->data;
-    return nullptr;
-}
-
-SlsMat SlsMatWrapper::slsmat() const {
-    return matrix;
-}
-
-
-
-DlsMatWrapper::DlsMatWrapper(long int M, long int N)
-    : matrix(NewDenseMat(M, N))
+    
+SUNMatrixWrapper::SUNMatrixWrapper(int M, int N)
+: matrix(SUNDenseMatrix(M, N))
 {
-    if((M*N > 0) && !matrix)
```

Comment:
That gave me a compiler warning. Multiplication in boolean context or something like that.